iib/workers/tasks/fbc_utils.py

```diff
@@ -86,6 +90,7 @@ def merge_catalogs_dirs(src_config: str, dest_config: str):
             msg = f"config directory does not exist: {conf_dir}"
             log.error(msg)
             raise IIBError(msg)
+        enforce_json_config_dir(conf_dir)
```

Comment:
I am on the edge if we should enforce it even when both inputs are in yaml. 
Do you think that there might be situation when yaml format is preferred? 

Just to discuss every corner-case:
I mean team maintaining opted-in operators might be surprised when the catalog is converted to json.
I am not sure if they use the resulted image for some tests - is not here any team that is extracting the data put inside the index and expecting yaml?

Reply:
That's something I want to know as well: Does it make a huge difference in the operator PoV to have YAML or JSON files? Because if this matters then enforcing one format all over the place is really not good, but I'm not sure how OLM deals with this.

What could be done is to, like, verify the format of the first operator and enforce that format, I'm just not sure whether this would be overengineering or not...

Either way, if the format type for OLM operators really matters, instead of just the content, I may update the `opm migrate -o {FORMAT}` in IIB but I agree discussing this with more people would be ideal

Reply:
@xDaile @chandwanitulsi @yashvardhannanavati thoughts?

Reply:
Very valid question. 
AFAIK product teams use OLM to test/use their operators and OLM will support both json and yaml format without any differences to user so we should be good here. But its possible to have outliers in product teams, I have reached out to Jordan to verify whether we are expected to have any specific format of files in index image based on inputs, let's see what he suggests

Reply:
Based on the suggestion, I don't think product teams would check the content and expect a certain format of catalog file in the index, so we can go ahead with this